srcalc/src/main/java/gov/va/med/srcalc/vista/RpcVistaPatientDao.java

```diff
@@ -162,4 +170,53 @@ private void parseRecentVitalResults(final Patient patient, final List<String> v
     	patient.setBmi(Double.parseDouble(bmiLineTokens[bmiLineTokens.length-2]));
     	patient.setBmiDate(patient.getWeightDate());
     }
+
+	@Override
+	public String saveRiskCalculationNote(final Calculation calculation,
+			final String electronicSignature) {
+		final StringBuilder noteBody = buildNoteBody(calculation);
+		try 
+		{
+			final List<String> saveResults;
+			saveResults = fProcedureCaller.doRpc(
+	            fDuz, RemoteProcedure.SAVE_PROGRESS_NOTE,
+	            fDuz, electronicSignature, String.valueOf(calculation.getPatient().getDfn()), noteBody.toString());
+			final String[] splitArray = saveResults.get(0).split("\\^");
+			if(splitArray[0].equals("1"))
+			{
+				return SUCCESS;
+			}
+			else
+			{
+				return INVALID_SIGNATURE;
+			}
+		}
+		catch(final Exception e)
+		{
+			// An Exception means an invalid DUZ or a problem with VistALink/VistA
+			// Translate the exception into a status message
+			return VISTA_EXCEPTION_STATUS;
```

Comment:
Why not throw the exception? Indicating failure via return codes is so 20 years ago.